File: Java/String-2/xyzThere.java

public class string2 {
	
	// return true if the given string contains an appearance of "xyz" where the xyz is not directly preceeded by a period (.)
	// so "xxyz" counts but "x.xyz" does not
	public boolean xyzThere(String str) {
		
		boolean b = false;

		for (int i = 0; i < str.length()-2; i++) {
			if (i == 0 && str.charAt(i) == 'x' && str.substring(i, i+3).equals("xyz")) {
			  b = true;
			}
			else if (i > 0 && str.charAt(i-1) != '.' && str.charAt(i) == 'x' && str.substring(i, i+3).equals("xyz")) {
				b = true;
			}
		}
		
		return b;
		
	}

	public static void main(String[] args) {

		string2 o = new string2();
		
		System.out.println(o.xyzThere("abcxyz"));
		System.out.println(o.xyzThere("abc.xyz"));
		System.out.println(o.xyzThere("xyz.abc"));

	}

}

File: JavaScript/String-2/xyzThere.js

// return true if the given string contains an appearance of 'xyz' where the xyz is not directly preceeded by a period (.)
// so 'xxyz' counts but 'x.xyz' does not
const xyzThere = (str) => {
    if (str.length === 3 && str === 'xyz') {
        return true;
    }

    for (let i = 0; i < str.length - 3; i++) {
        if (i === 0 && str.substring(i, i + 3) === 'xyz') {
            return true;
        } else {
            if (str[i] !== '.' && str.substring(i + 1, i + 4) === 'xyz') {
                return true;
            }
        }
    }

    return false;
};

console.log(xyzThere('abcxyz'));
console.log(xyzThere('abc.xyz'));
console.log(xyzThere('xyz.abc'));
